tests/test_mark_tokens.py

```diff
@@ -448,3 +448,36 @@ def g(x):
     else:
       self.assertEqual(m.view_nodes_at(5, 0), {'FunctionDef:@deco2(a=1)\ndef g(x):\n  pass'})
     self.assertEqual(m.view_nodes_at(5, 1), {'Name:deco2', 'Call:deco2(a=1)'})
+
+  def test_with(self):
+    source = "with foo: pass"
+    m = self.create_mark_checker(source)
+    self.assertEqual(m.view_node_types_at(1, 0), {"Module", "With"})
+    self.assertEqual(m.view_nodes_at(1, 0), {
+      "Module:with foo: pass",
+      "With:with foo: pass",
+    })
+
+    source = textwrap.dedent(
+      '''
+      def f(x):
+        with A() as a:
+          log(a)
+          with B() as b, C() as c: log(b, c)
```

Comment:
Oh nice, I should have tested the multi-with case. Super interesting that it desugars it to be nested `with` nodes in Python 2, but a single `with` node with multiple items in Python 3 :)